templates/tower-compute-environment.j2

```diff
@@ -0,0 +1,283 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >-
+  Stack for provisioning the AWS resources needed for
+  configuring compute environments in Nextflow Tower.
+
+Parameters:
+  ReadWriteAccessArns:
+    Type: CommaDelimitedList
+    Default: ""
+    Description: >-
+      (Optional) List of IAM users and roles (strings) who
+      should have read- and write-access to the S3 bucket
+
+  ReadOnlyAccessArns:
+    Type: CommaDelimitedList
+    Default: ""
+    Description: >-
+      (Optional) List of IAM users and roles (strings)
+      who should only have read-access to the S3 bucket
+
+  AdminRoleArns:
+    Type: CommaDelimitedList
+    Description: >-
+      List of IAM role ARNs (strings) with administrator
+      access and should retain access to restricted resources
+      (e.g., private S3 bucket, KMS key, secrets)
+
+  TowerForgePolicyArn:
+    Type: String
+    Description: ARN for Tower Forge IAM policy
+
+  TowerLaunchPolicyArn:
+    Type: String
+    Description: ARN for Tower Launch IAM policy
+
+  TemplateRootUrl:
+    Type: String
+    Description: URL of S3 bucket where templates are deployed
+    ConstraintDescription: Must be a valid S3 HTTP URL
+
+Conditions:
+  HasReadWriteAccessArns:
+    !Not [!Equals [!Join [",", !Ref ReadWriteAccessArns], ""]]
+  HasReadOnlyAccessArns:
+    !Not [!Equals [!Join [",", !Ref ReadOnlyAccessArns], ""]]
+  HasAnyAccessArns:
+    !Or [!Condition HasReadWriteAccessArns, !Condition HasReadOnlyAccessArns]
+
+Resources:
+  ReadWriteAccessRole:
+    Type: AWS::IAM::Role
+    Condition: HasReadWriteAccessArns
+    Properties:
+      RoleName: !Sub "${AWS::StackName}-ReadWriteAccessRole"
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonS3FullAccess
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              AWS: !Ref ReadWriteAccessArns
+            Action:
+              - "sts:AssumeRole"
+
+  ReadOnlyAccessRole:
+    Type: AWS::IAM::Role
+    Condition: HasReadOnlyAccessArns
+    Properties:
+      RoleName: !Sub "${AWS::StackName}-ReadOnlyAccessRole"
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              AWS: !Ref ReadOnlyAccessArns
+            Action:
+              - "sts:AssumeRole"
+
+  EncryptionKeyStack:
+    Type: AWS::CloudFormation::Stack
```

Comment:
This is my suggestion because i feel that templates are more maintainable when not using nested stacks however that is only a preference, one that we discuss further in slack.  You can take my suggestion or do as you wish.